src/wallet/crypter.cpp

```diff
@@ -185,24 +185,22 @@ static bool DecryptSaplingSpendingKey(const CKeyingMaterial& vMasterKey,
     return sk.expsk.full_viewing_key() == extfvk.fvk;
 }
 
+// cs_KeyStore lock must be held by caller
```

Comment:
This function could `AssertLockHeld(cs_KeyStore)` but I think that adds production runtime overhead, doesn't it? (Not just debug build?)

Reply:
Since we're removing a lock acquisition here, the AssertLockHeld won't add more overhead than that would have had.